sys/dev/ti/if_ti.c

```diff
@@ -592,12 +594,12 @@ ti_copy_mem(struct ti_softc *sc, uint32_t tigon_addr, uint32_t len,
 					    sc->ti_membuf2, segsize,
 					    TI_SWAP_NTOH);
 					TI_UNLOCK(sc);
-					bcopy(&sc->ti_membuf2[segresid], ptr,
+					bcopy(&sc->ti_membuf2[segresid], (void *)(__cheri_addr uintptr_t)ptr,
```

Comment:
`(void *)(__cheri_addr uintptr_t)ptr` will strip the tag in the purecap kernel.
```suggestion
					bcopy(&sc->ti_membuf2[segresid], (__cheri_fromcap void *)ptr,
```

Reply:
Do we really want this to be kernel-ddc relative?

Reply:
Well, making it a `void *` in hybrid is going to be kernel DDC relative anytime it is dereferenced.  Perhaps it should use fromcap?  I'd have to fix the other end of it to use tocap though instead of whatever it does now perhaps to stuff a kernel pointer into the capability to pass into this.

Reply:
Ended up using `__cheri_fromcap`